iothub/device/src/IotHubDeviceClient.cs

```diff
@@ -109,6 +109,22 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         public void SetConnectionStateChangeHandler(Action<ConnectionState, ConnectionStateChangeReason> stateChangeHandler)
             => InternalClient.SetConnectionStateChangeHandler(stateChangeHandler);
 
+        /// <summary>
+        /// Last connection state.
+        /// </summary>
+        public ConnectionState LastConnectionState
+        {
+            get => InternalClient._lastConnectionState;
+        }
+
+        /// <summary>
+        /// Last connection state change reason.
+        /// </summary>
+        public ConnectionStateChangeReason LastConnectionStateChangeReason
+        {
+            get => InternalClient._lastConnectionStateChangeReason;
+        }
```

Comment:
I'd rather expose this as a class that has cohesive properties.

e.g. `IotHubDeviceClient.LatestConnectionState` which would have properties `State`, `ChangeReason`, and `ChangedDateTimeUtc`.

Reply:
> I'd rather expose this as a class that has cohesive properties.
> 
> e.g. `IotHubDeviceClient.LatestConnectionState` which would have properties `State`, `ChangeReason`, and `ChangedDateTimeUtc`.

Yeah, I planned to do that in another PR originally, but sure, I can also do that here

Reply:
Fine, I'm okay with that, but it seems like more work to make such a small number of changes in 2 PRs. Up to you.